core/src/main/java/Scenes/Leaderboards.java

```diff
@@ -60,15 +52,25 @@ public void render(float delta) {
     spriteBatch.end();
   }
 
-  @Override
-  public boolean tap(float x, float y, int count, int button) {
-    float worldX = x * WidthWorldPixelRatio;
-    float correctedY = Gdx.graphics.getHeight() - y;
-    float worldY = correctedY * HeightWorldPixelRatio;
-    if (backButton.isClicked(worldX, worldY)) {
-      System.out.println("Go back to main menu");
-      mainGame.setScreen(FilePlayMain.ScreenType.MAINMENU);
-    }
-    return false;
+  /**
+   * This adds listeners to each of the buttons along with what to do upon being touched. A new
+   * InputListener is added inline overriding the touchDown method to determine what to do when
+   * touched.
+   */
+  private void addAllListeners() {
```

Comment:
We will not be able to call addAllListeners() and addAllActors() in the BaseScreen class since they are declared abstract there. Java throws a NullException if I do. I can still add the abstract declarations in the BaseScreen class but they will still need to be defined and called from each subclass.